src/pages/resultsView/ResultsViewPageStoreUtils.ts

```diff
@@ -805,6 +809,45 @@ function sortEnrichmentData(data: any[]): any[] {
     return _.sortBy(data, ['pValue', 'hugoGeneSymbol']);
 }
 
+export function makeGenericAssayEnrichmentDataPromise<
+    T extends {
+        stableId: string;
+        pValue: number;
+        qValue?: number;
+    }
+>(params: {
+    resultViewPageStore?: ResultsViewPageStore;
+    await: MobxPromise_await;
+    getSelectedProfileMap: () => { [studyId: string]: MolecularProfile };
+    fetchData: () => Promise<T[]>;
+}): MobxPromise<(T & { qValue: number })[]> {
+    return remoteData({
+        await: () => {
+            const ret = params.await();
+            if (params.resultViewPageStore) {
+                ret.push(params.resultViewPageStore.selectedMolecularProfiles);
+            }
+            return ret;
+        },
+        invoke: async () => {
+            const profileMap = params.getSelectedProfileMap();
+            if (profileMap) {
+                let data = await params.fetchData();
+                return calculateQValuesAndSortEnrichmentData(
+                    data,
+                    sortGenericAssayEnrichmentData
+                );
+            } else {
+                return [];
+            }
+        },
+    });
+}
+
+function sortGenericAssayEnrichmentData(data: any[]): any[] {
```

Comment:
my guess is this could be typed?   i.e. pValue and stableId are members of some type?

Reply:
Yes, it can be typed. I will give it the type `GenericAssayEnrichmentWithQ` to it.